File: frameworks/shared/php/src/ProbeType.php
<?php

namespace VelocityBench\HealthCheck;

/**
 * Health check probe types (Kubernetes-compatible).
 */
enum ProbeType: string
{
    case LIVENESS = 'liveness';
    case READINESS = 'readiness';
    case STARTUP = 'startup';

    public static function fromString(string $value): self
    {
        return match (strtolower($value)) {
            'liveness' => self::LIVENESS,
            'readiness' => self::READINESS,
            'startup' => self::STARTUP,
            default => throw new \InvalidArgumentException("Unknown probe type: $value"),
        };
    }
}

File: frameworks/shared/java/src/main/java/com/velocitybench/healthcheck/ProbeType.java
package com.velocitybench.healthcheck;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Health check probe types (Kubernetes-compatible).
 */
public enum ProbeType {
    LIVENESS("liveness"),
    READINESS("readiness"),
    STARTUP("startup");

    private final String value;

    ProbeType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    public static ProbeType fromString(String value) {
        for (ProbeType type : ProbeType.values()) {
            if (type.value.equalsIgnoreCase(value)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown probe type: " + value);
    }
}
